### ISSUE TITLE
Crash occurs after trying to apply PBR material to one face of a cube

### ISSUE THREAD

**Issue by sl-service-account** [secondlife/viewer#2428]
**Steps to reproduce:**
1. Login to the [viewer](https://github.com/secondlife/viewer/releases/tag/Second_Life_Release%231adf2d0-DeltaFPS).
2. Rez a cube > switch build tools to 'Select Face'.
3. Click any face of the cube.
4. Go to the 'Texture' tab > select 'PBR Metallic roughtness' in Material dropdown.
5. Select any material (no matter what way - using Material picker or 'Choose from inventory' button).
6. Click 'OK' button in 'Pick: texture' floater.

**Actual result:** The viewer crashes.

**Expected result:** The viewer should not crash.

**Additional info:**
The issue was found during https://github.com/secondlife/viewer/issues/1922 testing.
The issue is not reproduced on the current release viewer (7.1.9.10515727195).
The issue is not reproduced after applying PBR material to a cube without 'Select Face' option.

The issue is reproduced on Win10/OSX:
[Win10 BugSplat Key Crash 95817](https://app.bugsplat.com/v2/keycrash?stackKeyId=95817&database=SecondLife_Viewer_2018)
[OSX BugSplat Key Crash 95819](https://app.bugsplat.com/v2/keycrash?stackKeyId=95819&database=SecondLife_Viewer_2018)

<details><summary>Win10 environment</summary>
<p>

Second Life Release 7.1.10.10516785745 (64bit)
Release Notes

You are at 128.4, 84.3, 21.8 in Product Engine 2 located at simhost-0353d13a62153db1f.agni
SLURL: http://maps.secondlife.com/secondlife/Product%20Engine%202/128/84/22
(global coordinates 337,024.0, 307,284.0, 21.8)
Second Life Server 2024-07-22.10048683488
Release Notes

CPU: 11th Gen Intel(R) Core(TM) i7-11370H @ 3.30GHz (3302.4 MHz)
Memory: 16076 MB
OS Version: Microsoft Windows 10 64-bit (Build 19045.4780)
Graphics Card Vendor: NVIDIA Corporation
Graphics Card: NVIDIA GeForce RTX 3060 Laptop GPU/PCIe/SSE2

Windows Graphics Driver Version: 32.0.15.6094
OpenGL Version: 4.6.0 NVIDIA 560.94

Window size: 1920x1027
Font Size Adjustment: 96pt
UI Scaling: 1
Draw distance: 128m
Bandwidth: 3000kbit/s
LOD factor: 1.75
Render quality: 5
Texture memory: 5228MB
Disk cache: Max size 1638.4 MB (7.7% used)

J2C Decoder Version: KDU v7.10.4
Audio Driver Version: OpenAL, version 1.1 ALSOFT 1.23.1 / OpenAL Community / OpenAL Soft: OpenAL Soft
Dullahan: 1.14.0.202408091639
  CEF: 118.4.1+g3dd6078+chromium-118.0.5993.54
  Chromium: 118.0.5993.54
LibVLC Version: 3.0.21
Voice Server Version: Vivox 4.10.0000.32327.5fc3fe7c.5942f08

Packets Lost: 0/7,607 (0.0%)
August 27 2024 07:01:04

</p>
</details> 

<details><summary>OSX environment</summary>
<p>

Second Life Release 7.1.10.10516785745 (64bit)
Release Notes

You are at 128.4, 84.3, 21.8 in Product Engine 2 located at simhost-0353d13a62153db1f.agni
SLURL: http://maps.secondlife.com/secondlife/Product%20Engine%202/128/84/22
(global coordinates 337024.0, 307284.0, 21.8)
Second Life Server 2024-07-22.10048683488
Release Notes

CPU: Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz (2200 MHz)
Memory: 16384 MB
OS Version: macOS 12.6.0 Darwin 21.6.0 Darwin Kernel Version 21.6.0: Mon Aug 22 20:17:10 PDT 2022; root:xnu-8020.140.49~2/RELEASE_X86_64 x86_64
Graphics Card Vendor: ATI Technologies Inc.
Graphics Card: AMD Radeon Pro 555X OpenGL Engine

OpenGL Version: 4.1 ATI-4.8.101

Window size: 1680x920
Font Size Adjustment: 96pt
UI Scaling: 1
Draw distance: 128m
Bandwidth: 3000kbit/s
LOD factor: 1.25
Render quality: 2
Texture memory: 4096MB
Disk cache: Max size 1638.4 MB (0.6% used)
HiDPI display mode: 

J2C Decoder Version: KDU v7.10.4
Audio Driver Version: OpenAL, version 1.1 ALSOFT 1.23.1 / OpenAL Community / OpenAL Soft: OpenAL Soft
Dullahan: 1.14.0.202408091638
  CEF: 118.4.1+g3dd6078+chromium-118.0.5993.54
  Chromium: 118.0.5993.54
LibVLC Version: 3.0.21
Voice Server Version: Vivox 4.10.0000.32327.5fc3fe7c.5942f08

Packets Lost: 0/501 (0.0%)
August 27 2024 07:11:56

</p>
</details>

**Comment by RunitaiLinden** [comment]
Fixed in DeltaFPS

There was a missing null check on a vertex buffer pointer.

**Comment by sergiiiproductengine** [comment]
The issue was reproduced on the DeltaFPS RC Second Life Release 7.1.10.10582490681 on Wn10 and Win11 with another call stack:

https://app.bugsplat.com/v2/keycrash?database=SecondLife_Viewer_2018&stackKeyId=95862.

**Comment by sergiiiproductengine** [comment]
Failed QA.

The issue is reproducible with all prims except cube on OSX (prism, pyramid, cylinder, sphere, etc):
https://app.bugsplat.com/v2/keycrash?database=SecondLife_Viewer_2018&stackKeyId=95819

The issue is still reproducible with all prims on Win10/Win11:
https://app.bugsplat.com/v2/keycrash?database=SecondLife_Viewer_2018&stackKeyId=95862

Verified on the Second Life Release 7.1.10.10582490681 on Win10/Win11/OSX in the scope of https://github.com/secondlife/iqa/issues/316.

**Comment by RunitaiLinden** [comment]
Fixed.

Embarrassingly missed another dereference just above the spot where I added a null check...

More details:

Root cause is this change: https://github.com/secondlife/viewer/commit/17f7b439d617bfdf4c865f1478ade662e86b61d2#diff-5d51e99abe74be94ea64daa3bc6d7e52e3f71538d9dfa5b988b426cde284d4bd

There, mVertexBufferGLTF was added to LLFace to fix #1922.  This is probably not a good long term fix, but refactor is low priority unless it shows up in profiles.  Added a TODO comment to factor out mVertexBufferGLTF.
